torchci/lib/fetchFlakyTests.ts

```diff
@@ -74,3 +74,192 @@ export async function fetchFlakyTestsAcrossJobs(
   );
   return flakyTestQuery.results ?? [];
 }
+
+export async function fetchFlakyTestsAcrossFileReruns(
+  numHours: string = "3"
+): Promise<FlakyTestData[]> {
+  const rocksetClient = getRocksetClient();
+  const failedTestsQuery = `
+select
+  DISTINCT
+  t.name,
+  t.file,
+  t.invoking_file,
+  t.classname,
+from
+  commons.test_run_s3 t
+where
+  t._event_time > CURRENT_TIMESTAMP() - HOURS(:numHours)
+  and (
+      t.failure is not null
+      or t.error is not null
+  )
+  and t.file is not null
+`;
+
+  const checkEveryTestQuery = `
+select
+    t.name,
+    t.classname as suite,
+    t.file,
+    t.invoking_file,
+    t.job_id,
+    1 as numGreen,
+    SUM(
+        if (
+            t.failure is null,
+            if(TYPEOF(t.rerun) = 'object', 1, Length(t.rerun)),
+            if(TYPEOF(t.rerun) = 'object', 2, Length(t.rerun) + 1)
+        )
+    ) as numRed
+FROM
+    commons.test_run_s3 t
+where
+    t.name = :name
+    and t.classname = :classname
+    and t.invoking_file = :invoking_file
+    and t.file = :file
+    and t._event_time > CURRENT_TIMESTAMP() - HOURS(:numHours)
+GROUP BY
+    name,
+    suite,
+    file,
+    invoking_file,
+    job_id
+HAVING
+    BOOL_OR(t.failure is null)
+    and BOOL_OR(t.failure is not null or t.error is not null)
+`;
+
+  const workflowJobInfoQuery = `
+select
+  j.name,
+  w.name as workflow_name,
+  j.id,
+  w.id as workflow_id,
+  w.head_branch,
+  j.run_attempt,
+  j.html_url
+from
+  workflow_job j join workflow_run w on w.id = j.run_id
+where
+  ARRAY_CONTAINS(SPLIT(:job_ids, ','), CAST(j.id as STRING))
+`;
+
+  // Get every distinct failed test on master in the past numHours (usually not a lot)
+  const failedTests = await rocksetClient.queries.query({
+    sql: {
+      query: failedTestsQuery,
+      parameters: [
+        {
+          name: "numHours",
+          type: "int",
+          value: numHours,
+        },
+      ],
+    },
+  });
+  let failedTestsResults = failedTests.results ?? [];
+
+  // For every failed test, query rockset for jobs that had file level reruns of
+  // the test in the past numHours.  Do this separately because a join on
+  // test_run_s3 takes a long time.  Batch the query since rockset doesn't allow
+  // more tha 150 concurrent queries.  Flatten the accumulator since it ends up
+  // being an array of arrays.
+  let rerunTestsUnflattened: any[] = [];
```

Comment:
Do you think if there are any duplication among `flakyTests`, `flakyTestsAcrossJobs`, and`flakyTestsAcrossFileReruns`?  If yes, I think you could exclude those tests that have already been covered by the first two cases from `failedTestsResults`.  Then `flakyTestsAcrossFileReruns` would be smaller because it doesn't need to include all test failures, which helps.

Reply:
I think there might be a few but most of the extra tests are probably going to be tests that are true failures